Internal/Impl.as

```diff
@@ -62,6 +71,23 @@ namespace VehicleState
 		}
 #endif
 
+#if TMNEXT
+		if (vis is null) {
+			auto gameTerminal = GetGameTerminal();
+			if (gameTerminal !is null) {
+				auto type = Reflection::GetType("CGameTerminal");
+				if (type !is null) {
+					auto offset = type.GetMember("MediaAmbianceClipPlayer").Offset + 0x68;
+					bool isWatchingGhost = Dev::GetOffsetUint8(gameTerminal, offset) > 0;
+					auto ghostVisEntId = Dev::GetOffsetUint32(gameTerminal, offset + 0x4);
+					if (isWatchingGhost && ghostVisEntId & 0x04000000 > 0) {
+						@vis = GetVis(sceneVis, ghostVisEntId);
+					}
+				}
+			}
+		}
+#endif
+
```

Comment:
Couple of problems here:
1. Types and member offsets are not cached, this causes unnecessary overhead
2. This offset of `+ 0x68` is in my opinion too large to use reliably as it will introduce a lot of new maintenance work - we should consider requesting Nadeo for a native property

Reply:
> Types and member offsets are not cached, this causes unnecessary overhead

Yep, obvs easily fixable.

> This offset of + 0x68 is in my opinion too large to use reliably as it will introduce a lot of new maintenance work - we should consider requesting Nadeo for a native property

Yeah okay. I agree it's large enough that it might get updated, but I disagree with the maintenance cost. 

- There is no chance of a crash since the offset is well within CGameTerminal's in-memory size and no pointers are followed.
- There are 2 checks (note: updated the first one), the bool == 1 and the 0x04000000 bit is set on the 'entity ID'. even after that, nothing will be returned from the other methods if no ent id matches.

So the worth that happens is the feature stops working. In rare cases it might use an incorrect ent ID. 

The overhead for updating the offset is just opening the game terminal in cheat engine and watching for somewhere near that offset that changes in the way that's expected, which is like a 5 minute job. 

I'm not sure if I'm missing anything about why it would be high-overhead tho (wrt maintenance). I'm not counting openplanet releases, tho, so mb that's it?

Reply:
Oh I agree with asking nadeo for an exposed property btw, that is easily the best solution

Reply:
The biggest maintenance overhead is actually finding out that it is broken. We currently do not have any automated tests in place to detect when these things are broken (unlike some built-in Openplanet stuff). It would be nice to get some kind of test suite for all of these dev things so they can be caught early whenever a game update comes out.